src/Microsoft.AspNetCore.Razor/Parser/TagHelpers/TagHelperParseTreeRewriter.cs

```diff
@@ -353,10 +427,15 @@ private IEnumerable<string> GetAttributeNames(Block tagBlock)
                     }
                 }
 
-                attributeNames.Add(childSpan.Content.Substring(start, end - start));
+                var attributeName = childSpan.Content.Substring(start, end - start);
+                var attributeValue = _attributeValueBuilder.ToString();
+                var attribute = new KeyValuePair<string, string>(attributeName, attributeValue);
+                attributes.Add(attribute);
+
+                _attributeValueBuilder.Clear();
```

Comment:
No, meant the two `Clear()` calls (or the general concept of returning the system to its initial state) should be together.